hnn_core/extracellular.py

```diff
@@ -0,0 +1,601 @@
+            Amount to offset traces by on the voltage-axis. Useful for plotting
+            laminar arrays.
+        voltage_scalebar : float | None (optional)
+            Height, in units of uV, of a scale bar to plot in the top-left
+            corner of the plot.
+        contact_labels : list (optional)
+            Labels associated with the contacts to plot. Passed as-is to
+            :meth:`~matplotlib.axes.Axes.set_yticklabels`.
+        show : bool
+            If True, show the figure
+
+        Returns
+        -------
+        fig : instance of plt.fig
+            The matplotlib figure handle into which time series were plotted.
+        """
+        from .viz import plot_extracellular
+
+        if trial_no is None:
+            plot_data = self.get_data()
+        elif isinstance(trial_no, (list, tuple, int, slice)):
+            plot_data = self.get_data()[trial_no, ]
+        else:
+            raise ValueError(f'unkown trial number type, got {trial_no}')
+
+        if isinstance(contact_no, (list, tuple, int, slice)):
+            plot_data = plot_data[:, contact_no, ]
+        elif contact_no is not None:
+            raise ValueError(f'unkown contact number type, got {contact_no}')
+
+        for trial_data in plot_data:
+            fig = plot_extracellular(
+                self.times, trial_data, tmin=tmin, tmax=tmax, ax=ax,
+                decim=decim, color=color,
+                voltage_offset=voltage_offset,
+                voltage_scalebar=voltage_scalebar,
+                contact_labels=contact_labels,
+                show=show)
+        return fig
+
+
+class ExtracellularArrayBuilder(object):
+    """The ExtracellularArrayBuilder class
+
+    Parameters
+    ----------
+    array : ExtracellularArray object
+        The instance of :class:`hnn_core.extracellular.ExtracellularArray` to
+        build in NEURON-Python
+    """
+    def __init__(self, array):
+        self.array = array
+        self.n_contacts = array.n_contacts
+        self._nrn_imem_ptrvec = None
+        self._nrn_imem_vec = None
+        self._nrn_r_transfer = None
+        self._nrn_times = None
+        self._nrn_voltages = None
+
+    def _build(self, cvode=None):
+        """Assemble NEURON objects for calculating extracellular potentials.
+
+        The handler is set up to maintain a vector of membrane currents at at
+        every inner segment of every section of every cell on each CVODE
+        integration step. In addition, it records a time vector of sample
+        times.
+
+        Parameters
+        ----------
+        cvode : instance of h.CVode
+            Multi order variable time step integration method.
+        """
+        # ordered list of h.Sections on this rank (if running in parallel)
+        secs_on_rank = _get_sections_on_this_rank()
+        # np.array of number of segments for each section, ordered as above
+        segment_counts = np.array(_get_segment_counts(secs_on_rank))
```

Comment:
Somewhat nitpicky, but I would just replace this with the one-liner list enumeration. So folks don't have to scroll up. Then you don't have to give the explanation:

```py
# np.array of number of segments for each section, ordered as above
```

because python reads like English when written well ;-)